### ISSUE TITLE
Warning when opening a URL

### ISSUE THREAD

**Issue by qwertynik** [birdofpreyru/react-native-static-server#63]
When opening a URL of a HTML file stored on the device, a warning is printed
```
 WARN  Error opening URL:  [Error: Unable to open URL: file:///Users/inqutec/Library/Developer/CoreSimulator/Devices/FCF07C4B-7AAA-48B2-9A39-6C4ADA268813/data/Containers/Bundle/Application/A44D9AFF-AFC6-4925-8859-199116E60074/LocalRender.app/null/local.html]
```
This is coming from  WebViewShared.js:42. Did not find a JS variant in the repo. Looks like this is a compile time file. 
Warning image:
<img width="914" alt="Screenshot 2023-08-09 at 6 02 27 PM" src="https://github.com/birdofpreyru/react-native-static-server/assets/6836536/9bab901f-9161-4edc-8e7b-01613948bec1">

Although the HTML on device loads, and the warning does not break any feature, wanted to know how this warning can be eliminated.

**Local setup info**:
Device: Mac M1
Node package.json:
```
  "dependencies": {
    "@dr.pogodin/react-native-static-server": "^0.7.13",
    "@react-native-community/cli-doctor": "^11.3.5",
    "react": "18.2.0",
    "react-native": "0.72.1",
    "react-native-fs": "^2.20.0",
    "react-native-webview": "^13.2.2"
  }
```

**Comment by birdofpreyru** [comment]
In the warning message you have `null` as one part of the path. I guess, you did some mistake, and trying to load the page before the server / path are ready?

**Comment by qwertynik** [comment]
That's right. Had noticed the `null` in the path myself, however, for other cases as well had noticed this warning. Is there anything to check to see if the server is ready other than waiting for `.start()` function call?

Can you elaborate on what is meant by path being ready?

**Comment by birdofpreyru** [comment]
> _Had noticed the null in the path myself, however, for other cases as well had noticed this warning._

There is no similar warning in the Example app included into this repo, thus it is some mistake in your code. 

> _Is there anything to check to see if the server is ready other than waiting for .start() function call?_

Yes, you can connect event listeners to a server instance, to get notified about its state changes &mdash;
see [.addStateListener()](https://github.com/birdofpreyru/react-native-static-server#addstatelistener) method.

> _Can you elaborate on what is meant by path being ready?_

I just meant, your inserted into the path some variable (server origin?) which was still _null_ at that moment.

**Comment by qwertynik** [comment]
@birdofpreyru Found the source of the problem. The issue is with whitelisting. Is there a section in README.md listing common errors? Will create a PR for the same.

**Comment by qwertynik** [comment]
Of course, in case you managed to solve the problem differently can you share the steps.

**Comment by birdofpreyru** [comment]
I'll repeat &mdash; I don't see any problem with the library here; I only see you are trying to engage me into a free work on your project.

**Comment by qwertynik** [comment]
> I'll repeat — I don't see any problem with the library here; 

I had not concluded that the library had an issue. On encountering the issue, reported it to find ways to resolve it. This could have either led to finding an issue in the library, or, the code. On further debugging realized that the warning can be resolved by using whitelisting - which might be a hack, but sufficient for now. However, still uncertain about the source of the warning. In this case, an HTML is downloaded and rendered from the storage, a use-case different from what's present in the Example app, so looking into this issue might lead to finding problems.

>  I only see you are trying to engage me into a free work on your project.

Not certain why you felt this way. However, your responses earlier on other issues and the library itself has certainly been useful.

**Comment by qwertynik** [comment]
In case attaching the code made you feel as if free dev service for the project is being sought, it can be ignored. It was attached to make understanding logs simpler - had thought about removing the code yesterday, but let it remain for clarity. Looking at the logs alone might point to issues, if any.

**Reattaching here:**
From the logs, it can be seen that the fileDir is set as "/Users/inqutec/Library/Developer/CoreSimulator/Devices/FCF07C4B-7AAA-48B2-9A39-6C4ADA268813/data/Containers/Data/Application/91D63B5C-87C1-4217-A7D1-0172C5A1CD79/Documents/localTest" but the path in the warning is this: file:///Users/inqutec/Library/Developer/CoreSimulator/Devices/FCF07C4B-7AAA-48B2-9A39-6C4ADA268813/data/Containers/Bundle/Application/C59E0A2F-142F-488E-A99F-8E9730956DED/LocalRender.app/null/local.html

The initial portion of the path is same file:///Users/inqutec/Library/Developer/CoreSimulator/Devices/FCF07C4B-7AAA-48B2-9A39-6C4ADA268813/data/Containers/. However, after this, the path shown in the warning is Bundle/Application/C59E0A2F-142F-488E-A99F-8E9730956DED/LocalRender.app/null/local.html whereas the path in the fileDir variable (set when instantiating the server) is Data/Application/91D63B5C-87C1-4217-A7D1-0172C5A1CD79/Documents/localTest. 

This change of path is mostly happening in an interaction between the WebView and the static-server.
